src/Kiota.Generated/KiotaAppSettingsGenerator.cs

```diff
@@ -8,16 +8,16 @@
 public class KiotaAppSettingsGenerator : ISourceGenerator
 {
 
-    private string findFileAndRead(string directory, string name, int maxDepth) {
+    private static string findFileAndRead(string directory, string name, int maxDepth = 5) {
         try {
             using var reader = new StreamReader(Path.Join(directory, name), Encoding.UTF8);
             return reader.ReadToEnd();
-        } catch (FileNotFoundException e)
+        } catch (FileNotFoundException)
         {
             if (maxDepth > 0)
                 return findFileAndRead(Path.Join(directory, ".."), name, maxDepth - 1);
             else
-                throw e;
+                return string.Empty;
```

Comment:
The exception thrown is emitted only as a warning and is super useful to understand if/when something is going wrong, returning an empty string here is going to mask other issues.

I do believe that, if there is a bug with generation we should fix it.

Reply:
the thinking was that since treat warnings as errors was enabled in this PR, it's completely fail. But that's not the case so I reverted that change.